server/controller/course.controller.js

```diff
@@ -290,47 +124,32 @@ courseController.put(
   '/',
   passport.authenticate('jwt', { session: false }),
   validateCourseId,
-  async (req, res) => {
-    const errorsAfterValidation = validationResult(req);
-    if (!errorsAfterValidation.isEmpty()) {
-      res.status(400).json({
-        code: 400,
-        errors: errorsAfterValidation.mapped(),
-      });
-    } else {
-      try {
-        // Get course ObjectId from query
-        const course_id = req.query.course_id;
-
-        // Remove undefined properties
-        const updateData = removeUndefinedObjectProps(req.body);
+  (req, res) => {
+    validationHandler(req, res, async () => {
+      const { _id } = req.query;
+      const updateData = removeUndefinedObjectProps(req.body);
 
-        // Check if there is data to update
-        if (isObjectEmpty(updateData)) {
-          return res.status(400).json({ error: NO_DATA_TO_UPDATE });
-        }
-
-        const updatedCourse = await Course.findByIdAndUpdate(
-          { _id: course_id },
-          updateData,
-          { useFindAndModify: false, new: true }
-        );
+      if (isObjectEmpty(updateData))
+        return res.status(400).json({ error: NO_DATA_TO_UPDATE });
 
-        if (updatedCourse) {
-          res.status(200).json(updatedCourse);
-        } else {
-          generateServerErrorCode(
-            res,
-            403,
-            'update course error',
-            COURSE_DOES_NOT_EXIST,
-            'course_id'
-          );
+      return Course.findByIdAndUpdate(
```

Comment:
Will this be a problem that you are not calling **await** when you return the query call?

Reply:
I don't need to call await since I'll put a callback function after updating the data